src/client/Microsoft.Identity.Client/ApiConfig/DownStreamRestApi/PublicClientDownstreamApi.cs

```diff
@@ -0,0 +1,170 @@
+{
+    /// <summary>
+    /// Down stream api for public clients
+    /// </summary>
+    public class PublicClientDownstreamApi
+    {
+        private readonly int _maximumAuthenticationAttempts = 3;
+        string _resourceName;
+        DownstreamRestApiOptions _options;
+        IPublicClientApplication _pca;
+        private string _currentNonce = null;
+        //TODO, grab identity logger from application.
+        IIdentityLogger _logger = NullIdentityModelLogger.Instance;
+        private int _currentAuthetnicatinonAttempts = 0;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="resourceName"></param>
+        /// <param name="options"></param>
+        public PublicClientDownstreamApi(string resourceName, DownstreamRestApiOptions options)
+        {
+            _resourceName = resourceName;
+            _options = options;
+            _pca = PublicClientApplicationBuilder.CreateWithApplicationOptions(_options).Build();
+            //_logger = options.logger.
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="targetUrl"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public async Task<HttpResponseMessage> CallGetApiAsync(Uri targetUrl, CancellationToken cancellationToken = default)
+        {
+            bool isPopSupportedByResource = false;
+
+            if (_pca.IsProofOfPossessionSupportedByClient())
+            {
+                AuthenticationHeaderParser parsedHeaders = null;
+                try
+                {
+                    parsedHeaders = await AuthenticationHeaderParser.ParseAuthenticationHeadersAsync(targetUrl.AbsoluteUri).ConfigureAwait(false);
+                }
+                catch (MsalClientException)
+                {
+                    throw;
+                }
+
+                //Resource returned a pop header containing a nonce.
+                isPopSupportedByResource = parsedHeaders.Nonce != null;
+
+                if (!isPopSupportedByResource)
+                {
+                    _logger.Log(new LogEntry() { Message = "Proof-of-Possesion is not supported by resource." });
+                }
+                else
+                {
+                    _currentNonce = parsedHeaders.Nonce;
+                }
+            }
+
+            var result = await GetAuthResultAsync(targetUrl, _currentNonce).ConfigureAwait(false);
+
+            // At this point, we have either a POP or a Bearer token
+
+            // Make a call to the RP
+            // If we get 200 OK -> update the nonce from the AuthenticateInfo Header
+            // If we get 401 Unauthorized -> the nonce was probably expired, get the new one from the headers                
+            HttpResponseMessage response = await TryGetResponseFromResourceAsync(targetUrl.AbsoluteUri, result).ConfigureAwait(false);
+
+            _currentAuthetnicatinonAttempts++;
```

Comment:
there is a typo: `_currentAuthetnicatinonAttempts` => `_currentAuthenticationAttempts`